From 606f7d3ca2f5a0a5aa2a51f69904064759e7da2f Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 7 Jun 2024 14:29:59 +0200
Subject: [PATCH 0919/1695] drivers: stm32_rng: deprecate stm32_firewall usage

Deprecate the stm32_firewall usage and replace it with the use of
stm32_etzpc_check_ns_access() to check non-secure access to the RNG.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 20e4d3308f2370197d811bd98cf2c46460b1a8ef drivers: stm32_rng: deprecate stm32_firewall usage]
Change-Id: Ia13ab563e4e191accb1f6621b76aa82c431f5755
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/386039
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/drivers/stm32_rng.c | 30 +++++++++++++-----------------
 1 file changed, 13 insertions(+), 17 deletions(-)

diff --git a/core/drivers/stm32_rng.c b/core/drivers/stm32_rng.c
index 3241c8571..cf7726415 100644
--- a/core/drivers/stm32_rng.c
+++ b/core/drivers/stm32_rng.c
@@ -7,8 +7,11 @@
 #include <drivers/clk.h>
 #include <drivers/clk_dt.h>
 #include <drivers/rstctrl.h>
-#include <drivers/stm32_firewall.h>
+#include <drivers/stm32_etzpc.h>
 #include <drivers/stm32_rng.h>
+#if defined(CFG_STM32MP15)
+#include <drivers/stm32mp_dt_bindings.h>
+#endif /* defined(CFG_STM32MP15) */
 #include <io.h>
 #include <kernel/delay.h>
 #include <kernel/dt.h>
@@ -669,22 +672,15 @@ static TEE_Result stm32_rng_probe(const void *fdt, int offs,
 	if (res)
 		goto err;
 
-	if (IS_ENABLED(CFG_STM32MP15)) {
-		/*
-		 * Only STM32MP15 requires a software registering of RNG
-		 * secure state
-		 */
-		const struct stm32_firewall_cfg nsec_cfg[] = {
-			{ FWLL_NSEC_RW | FWLL_MASTER(0) },
-			{ }, /* Null terminated */
-		};
-		paddr_t pa = stm32_rng->base.pa;
-
-		if (stm32_firewall_check_access(pa, 0, nsec_cfg) == TEE_SUCCESS)
-			stm32mp_register_non_secure_periph_iomem(pa);
-		else
-			stm32mp_register_secure_periph_iomem(pa);
-	}
+#if defined(CFG_STM32MP15)
+	/*
+	 * Only STM32MP15 requires a software registering of RNG secure state
+	 */
+	if (!stm32_etzpc_check_ns_access(STM32MP1_ETZPC_RNG1_ID))
+		stm32mp_register_non_secure_periph_iomem(stm32_rng->base.pa);
+	else
+		stm32mp_register_secure_periph_iomem(stm32_rng->base.pa);
+#endif /* defined(CFG_STM32MP15) */
 
 	/* Power management implementation expects both or none are set */
 	assert(stm32_rng->ddata->has_power_optim ==
-- 
2.47.2

